src/kiwixmessagebox.cpp

```diff
@@ -45,3 +55,10 @@ void showInfoBox(QString title, QString text, QWidget *parent)
         dialog->deleteLater();
     });
 }
+
+KiwixMessageBox::Result showKiwixMessageBox(QString title, QString text, QWidget *parent, QString leftTitle, QString rightTitle)
+{
+    KiwixMessageBox *dialog = new KiwixMessageBox(title, text, false, parent, leftTitle, rightTitle);
+    return dialog->execDialog();
+}
```

Comment:
Is `dialog` somehow deleted automatically?

Reply:
Yes, along with the parent (when the parent - mainWindow here- is deleted).

Reply:
That's a memory leak because the memory allocated to each new (displayed and closed) dialog is not reclaimed until the end of the application.

Reply:
right right